f3-sidecar/ffi_impl.go

```diff
@@ -26,8 +26,12 @@ type f3Impl struct {
 }
 
 func (f3 *f3Impl) run(rpc_endpoint string, jwt string, f3_rpc_endpoint string, initial_power_table string, bootstrap_epoch int64, finality int64, db string, manifest_server string) bool {
-	err := run(f3.ctx, rpc_endpoint, jwt, f3_rpc_endpoint, initial_power_table, bootstrap_epoch, finality, db, manifest_server)
-	return err == nil
+	for {
+		err := run(f3.ctx, rpc_endpoint, jwt, f3_rpc_endpoint, initial_power_table, bootstrap_epoch, finality, db, manifest_server)
+		if err != nil {
+			logger.Errorf("Unexpected F3 failure, restarting... error=%s", err)
+		}
+	}
```

Comment:
If F3 is failing repeatedly for whatever reason it will spam the entire process with those fiascos and constant reboots, no?

I'd add a sleep in case of failures.

Reply:
Maximum number of fiascos might also be added so that we don't loop indefinitely.

Reply:
@LesnyRumcajs Good idea! Fixed.